contracts/utility/CarbonBatcher.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import { IERC721Receiver } from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+import { ICarbonController } from "../carbon/interfaces/ICarbonController.sol";
+import { IVoucher } from "../voucher/interfaces/IVoucher.sol";
+
+import { Upgradeable } from "./Upgradeable.sol";
+import { Order } from "../carbon/Strategies.sol";
+import { Utils } from "../utility/Utils.sol";
+import { Token } from "../token/Token.sol";
+
+struct StrategyData {
+    Token[2] tokens;
+    Order[2] orders;
+}
+
+/**
+ * @dev Contract to batch create carbon controller strategies
+ */
+contract CarbonBatcher is Upgradeable, Utils, ReentrancyGuard, IERC721Receiver {
+    using Address for address payable;
+
+    error InsufficientNativeTokenSent();
+
+    ICarbonController private immutable carbonController;
+    IVoucher private immutable voucher;
+
+    /**
+     * @dev triggered when tokens have been withdrawn from the carbon batcher
+     */
+    event FundsWithdrawn(Token indexed token, address indexed caller, address indexed target, uint256 amount);
+
+    constructor(
+        ICarbonController _carbonController,
```

Comment:
We don't prefix function arguments with an `_`, but if you did this in order to resolve the conflict with state variables, you can call rename these arguments to `initCarbonController`, and `initVoucher`, for example